server/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -958,10 +965,9 @@ private static long rawRelativeTimeInMillis() {
 
         private void close() {
             // attempt to close once atomically
-            if (closed.compareAndSet(false, true) == false) {
-                throw new IllegalStateException("Channel is already closed");
+            if (closed.compareAndSet(false, true)) {
+                inFlightRequestsBreaker(circuitBreakerService).addWithoutBreaking(-contentLength);
             }
```

Comment:
When client terminates request mid stream we need to release circuit-breaker. This `close` block can execute twice now, when client terminates connection. I dont see any harm in it.